File: src/org/espenhahn/allocate/likepaxos/registry/debug/mvc/animations/TranslateTextAnimation.java
package org.espenhahn.allocate.likepaxos.registry.debug.mvc.animations;

import java.util.function.Consumer;

import org.espenhahn.allocate.likepaxos.registry.debug.mvc.objects.Text;

public class TranslateTextAnimation extends TranslateAnimation<Text> {

	public TranslateTextAnimation(Text t, int end_x, int end_y, Consumer<Text> callback) {
		super(t, end_x, end_y, callback);
	}

	@Override
	protected void apply(int dx, int dy) {
		getDrawable().translate(dx, dy);
	}

}

File: src/org/espenhahn/allocate/likepaxos/registry/debug/mvc/animations/TranslateTextAnimation.kt
package org.espenhahn.allocate.likepaxos.registry.debug.mvc.animations

import org.espenhahn.allocate.likepaxos.registry.debug.mvc.objects.Text
import java.util.function.Consumer

class TranslateTextAnimation(t: Text, end_x: Int, end_y: Int, callback: Consumer<Text>) : TranslateAnimation<Text>(t, end_x, end_y, callback) {

    override fun apply(x: Int, y: Int) {
        drawable.translate(x, y)
    }

}
